sdk-java/src/main/java/io/littlehorse/sdk/worker/internal/LHTaskExecutor.java

```diff
@@ -0,0 +1,238 @@
+package io.littlehorse.sdk.worker.internal;
+
+import com.google.common.base.Throwables;
+import com.google.protobuf.Empty;
+import io.grpc.stub.StreamObserver;
+import io.littlehorse.sdk.common.LHLibUtil;
+import io.littlehorse.sdk.common.exception.InputVarSubstitutionError;
+import io.littlehorse.sdk.common.exception.LHSerdeError;
+import io.littlehorse.sdk.common.exception.LHTaskException;
+import io.littlehorse.sdk.common.proto.LHErrorType;
+import io.littlehorse.sdk.common.proto.LHTaskError;
+import io.littlehorse.sdk.common.proto.LittleHorseGrpc;
+import io.littlehorse.sdk.common.proto.LittleHorseGrpc.LittleHorseStub;
+import io.littlehorse.sdk.common.proto.ReportTaskRun;
+import io.littlehorse.sdk.common.proto.ScheduledTask;
+import io.littlehorse.sdk.common.proto.TaskStatus;
+import io.littlehorse.sdk.common.proto.VariableValue;
+import io.littlehorse.sdk.worker.WorkerContext;
+import io.littlehorse.sdk.worker.internal.util.VariableMapping;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Manages a queue and executor for scheduling and executing tasks in memory.
+ * Future versions will share the same queue for many workers.
+ * It handles retries with a specified delay.
```

Comment:
Could you please update this comment to be specific about what you mean by "retries"?

What I **think** this means is retries on the `rpc ReportTask`. Why do we need that?

Let's think about what we have going on here:
- We have a "bootstrap stub" which has access to all the servers.
- We have a "specific stub" which has access to one specific LH Server.
- When we do a `rpc ReportTask`, we use the Specific Stub so that it goes directly to the server that will process it. This saves us some network hops and is more performant.
- If the specific server is down, we retry with the Bootstrap Stub which has access to all the servers. This is useful for increasing availability during a server rolling restart.

Reply:
> We have a "bootstrap stub" which has access to all the servers.
That's not always the case, right? It depends on infrastructure decisions. 